### PR TITLE
Fix Preconditions

### PR DESCRIPTION
**Pull request by Dufgui**
Signed-off-by: Guillaume Dufour <guillaume.duff@gmail.com>

### CODE REVIEW

File: org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/NonNullTest.java [hunk 1]
```diff
@@ -12,8 +12,8 @@
 	public void testCodeAction1() {
 		try {
 			new CodeAction(null);
-			fail("Expected an IllegalArgumentException");
-		} catch (IllegalArgumentException exc) {
+			fail("Expected an NullPointerException");
```
Comment (KamasamaK) [inline org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/NonNullTest.java:15]: Grammar
```suggestion
			fail("Expected a NullPointerException");
```

File: org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/NonNullTest.java [hunk 2]
```diff
@@ -23,8 +23,8 @@ public void testCodeAction2() {
 		try {
 			CodeAction codeAction = new CodeAction();
 			codeAction.setTitle(null);
-			fail("Expected an IllegalArgumentException");
-		} catch (IllegalArgumentException exc) {
+			fail("Expected an NullPointerException");
```
Comment (KamasamaK) [inline org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/NonNullTest.java:26]: Grammar
```suggestion
			fail("Expected a NullPointerException");
```

**Comment by Dufgui** [comment]
thanks for the review @KamasamaK

**Comment by KamasamaK** [comment]
@eclipse-lsp4j-bot ok to test

**Comment by KamasamaK** [comment]
@eclipse-lsp4j-bot run tests

**Comment by KamasamaK** [comment]
I don't seem to be able to runs test. Maybe @jonahgraham or @spoenemann can do that.

I've run the tests locally with these changes and they pass. I guess this was caused by #431, which didn't have the tests run.

@nixel2007 Was this the expected result of your change?

**Comment by Dufgui** [comment]
> I don't seem to be able to runs test. Maybe @jonahgraham or @spoenemann can do that.
> 
> I've run the tests locally with these changes and they pass. I guess this was caused by #431, which didn't have the tests run.
> 
> @nixel2007 Was this the expected result of your change?

As you can read, guava use NullPointer and previous version on our PreConditions use IllegalArgumentException.
And when we migrate on our implementation we don't align our code to the guava behaviours.

https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java#L878

**Comment by nixel2007** [comment]
@KamasamaK

> @nixel2007 Was this the expected result of your change?

Guava-free components. :)

**Comment by jonahgraham** [comment]
> I don't seem to be able to runs test. Maybe @jonahgraham or @spoenemann can do that.


Hi @KamasamaK I added you as admin so you can trigger tests for others in https://ci.eclipse.org/lsp4j/job/lsp4j-github-pullrequests/configure - all committers can be in that list, I just don't know how to automate that.

**Comment by KamasamaK** [comment]
@eclipse-lsp4j-bot run tests

**Review by KamasamaK — APPROVED** [review]
LGTM

**Pull request merged**